library/include/flutter_desktop_embedding/glfw/flutter_window_controller.h

```diff
@@ -0,0 +1,89 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef LIBRARY_INCLUDE_FLUTTER_DESKTOP_EMBEDDING_GLFW_FLUTTER_WINDOW_CONTROLLER_H_
+#define LIBRARY_INCLUDE_FLUTTER_DESKTOP_EMBEDDING_GLFW_FLUTTER_WINDOW_CONTROLLER_H_
+
+#include <string>
+#include <vector>
+
+#include "embedder.h"
+
+#ifdef USE_FLATTENED_INCLUDES
+#include "fde_export.h"
+#include "plugin_registrar.h"
+#else
+#include "../fde_export.h"
+#include "../plugin_registrar.h"
+#endif
+
+namespace flutter_desktop_embedding {
+
+// A controller for a window displaying Flutter content.
+//
+// This is the primary wrapper class for the desktop embedding C API.
+// If you use this class, you should not call any of the setup or teardown
+// methods in embedder.h directly, as this class will do that internally.
+//
+// Note: This is an early implementation (using GLFW internally) which
+// requires control of the application's event loop, and is thus useful
+// primarily for building a simple one-window shell hosting a Flutter
+// application. The final implementation and API will be very different.
+class FDE_EXPORT FlutterWindowController {
+ public:
+  // There must be only one instance of this class in an application at any
+  // given time, as Flutter does not support multiple engines in one process,
+  // or multiple views in one engine.
```

Comment:
The reason this is just a comment rather an actual enforced singleton design is that my hope is that the C++ wrapper layer I'm working on extracting for #230 (which this would be part of) could be header-only, which wouldn't work with a singleton, so I'm structuring it this way to allow flipping to header-only without more API churn.